### DISCUSSION TITLE
Autocomplete cascading issue

### DISCUSSION THREAD

**Discussion by AndrzejKl** [Megabit/Blazorise#3535] [Q&A]
Hi
I'm trying to get functional autocomplete cascading, on first one I use OnSelectedValueChanged to load coresponding data to seconnd autocomplete, it works. But I'm strugling how to clear second autocomplete selectedValue and text, when first isn't choiced. It almost works (when I bind value second autocomplete to some property and clear it there, but in this case I got null exception for item.Name - in TextField (second autocomplete).
What's correct way to get fully functional cascading autocomplete?

**Comment by stsrki** [comment]
Seems that you happen to stumble on the same error as #3420

Generally, to clear the value you have the `await autocomplete.Clear()` method. But the recommended way is always to set the values through the parameters, in this case, `SelectedValue`.

**Reply by AndrzejKl** [reply]
Thank You for answer, 
Clear method works, at least I can go deeper, but seting SelectedValue via @bind-SelectedValue to 0 (in my case selectedvalue is ID column), cause null exception in TextField item.Name for item. Setting TextField=@((item) => item?.Name) doesn't help.
Thank you for this workaround.
